mobile/library/common/extensions/filters/http/network_configuration/filter.cc

```diff
@@ -64,8 +96,57 @@ NetworkConfigurationFilter::decodeHeaders(Http::RequestHeaderMap& request_header
     return Http::FilterHeadersStatus::Continue;
   }
 
-  // If there is no proxy configured, continue.
+  // TODO(Augustyniak): Update Android proxy resolution to use API extension registry. As of now,
+  // it's only iOS that uses that code path.
+  const auto proxy_resolver =
+      static_cast<envoy_proxy_resolver*>(Api::External::retrieveApiSafe("envoy_proxy_resolver"));
```

Comment:
iff we don't think we're migrating everything to the same path maybe make the naming here apple specific?

Reply:
kept the naming the same for now as we're unsure of the path forward and it's probably best that the network_configuration/filter.cc not know about an "apple-specific" API and just produce the general `envoy_proxy_resolver` API